src/Components/test/E2ETest/ServerRenderingTests/EnhancedNavigationTest.cs

```diff
@@ -114,7 +113,36 @@ public void EnhancedNavCanBeReenabledHierarchically()
         // Check we got there, and it did retain the <h1> element
         Browser.Equal("Other", () => Browser.Exists(By.TagName("h1")).Text);
         Assert.Equal("Other", originalH1Elem.Text);
+    }
+
+    [Fact]
+    public void EnhancedNavWorksInsideSVGElement()
+    {
+        Navigate($"{ServerPathBase}/nav");
+
+        var originalH1Elem = Browser.Exists(By.TagName("h1"));
+        Browser.Equal("Hello", () => originalH1Elem.Text);
+
+        Browser.Exists(By.TagName("nav")).FindElement(By.Id("svg-nav-link")).Click();
+
+        // Check we got there, and it did retain the <h1> element
+        Browser.Equal("Other", () => Browser.Exists(By.TagName("h1")).Text);
+        Assert.Equal("Other", originalH1Elem.Text);
+    }
```

Comment:
Ideally there'd also be a test for "works inside SVG element" for regular client-side (interactive) nav as well, since that was also not working before and is also fixed by your change, if I'm reading the code correctly.

Reply:
You are right. Click on anchor inside svg element doesn't get intercepted. I added the test